src/main/java/randoop/condition/PreconditionOutcomeTable.java

```diff
@@ -0,0 +1,138 @@
+ *       OperationConditions#checkThrowsPreconditions(java.lang.Object[])}.
+ *   <li>The expected postcondition, if any. If the preconditions are satisfied, test the guards of
+ *       the normal postconditions of the specification in order, and save the property for the
+ *       first guard satisfied, if there is one. See {@link
+ *       OperationConditions#checkPostconditionGuards(java.lang.Object[])}.
+ * </ol>
+ *
+ * <p>TODO: Merge the following in as well
+ *
+ * <ul>
+ *   <li>Whether the preconditions fail or are satisfied, given the observed values
+ *   <li>The set of expected exceptions, given the observed values (that is, which
+ *       throws-conditions' guards are satisfied)
+ *   <li>The expected postcondition, given the observed values (that is, which return-clauses'
+ *       guards are satisfied)
+ * </ul>
+ */
+public class PreconditionOutcomeTable {
+
+  /** Indicates whether this table is empty. */
+  private boolean isEmpty = true;
+
+  /** Indicates whether this table has a valid entry. */
+  private boolean hasValid = false;
+
+  /** The list of throws clauses for this table. */
+  private final List<Set<ThrowsClause>> exceptionSets;
+
+  /** The list of post-conditions for this table. */
+  private final List<PostCondition> postConditions;
+
+  /** Creates an empty {@link PreconditionOutcomeTable}. */
+  public PreconditionOutcomeTable() {
+    exceptionSets = new ArrayList<>();
+    postConditions = new ArrayList<>();
+  }
+
+  /**
+   * Adds the outcome of checking the conditions of a specification.
+   *
+   * @param preconditionsSatisfied boolean value indicating whether all preconditions are satisfied
+   * @param throwsClauses set of exception type-comment pairs for exceptions expected in post-state
+   * @param postCondition post-condition that must be true in post-state if no exception is thrown,
+   *     null if none
+   */
+  void add(
+      boolean preconditionsSatisfied,
+      Set<ThrowsClause> throwsClauses,
+      PostCondition postCondition) {
+    isEmpty = false;
+    if (preconditionsSatisfied) {
+      if (postCondition != null) {
+        postConditions.add(postCondition);
+      }
+      hasValid = true;
+    }
+    if (!throwsClauses.isEmpty()) {
+      exceptionSets.add(throwsClauses);
+    }
+  }
+
+  /**
+   * Indicate whether this set of results indicates a definitively invalid pre-state. Occurs when
+   * all preconditions fail and there are no expected exceptions.
+   *
+   * <p>This method should be called after all entries are added; that is, no more entries should be
+   * added after it is called.
+   *
+   * @return true if preconditions of all specifications are unsatisfied, and there are no expected
+   *     exceptions; false, otherwise
+   */
+  public boolean isInvalidPrestate() {
+    return !isEmpty && !hasValid && exceptionSets.isEmpty();
+  }
+
+  /**
+   * Constructs the {@link TestCheckGenerator} to apply post call by extending the given generator.
```

Comment:
Please make the specification say what the goal or purpose of the method is, from a client point of view. Saying it creates an object of a particular class is not helpful; a higher-level description is needed. Here, the goal is to check whether an invocation satisfies the specification.
The phrase "apply post call" is not clear.